osi_trafficcommand.proto

```diff
@@ -0,0 +1,507 @@
+// In the future this interface shall be aligned with the level of
+// atomic actions as described in OpenSCENARIO 1.0 or later.
+//
+message TrafficCommand
+{
+    // The interface version used by the sender (scenario engine).
+    //
+    optional InterfaceVersion version = 1;
+
+    // The data timestamp of the simulation environment. Zero time is arbitrary
+    // but must be identical for all messages. Zero time does not need to
+    // coincide with the UNIX epoch. Recommended is the starting time point of
+    // the simulation.
+    //
+    // \note For traffic command data the timestamp coincides both with
+    // the notional simulation time the data applies to and the time it was sent
+    // (there is no inherent latency for traffic command data, as opposed
+    // to sensor data).
+    //
+    optional Timestamp timestamp = 2;
+
+    // The ID of this traffic participant.
+    //
+    optional Identifier traffic_participant_id = 3;
+
+    // Commanded Traffic Action(s) if any
+    //
+    // \note OSI uses singular instead of plural for repeated field names.
+    //
+    // \note If more than one action is being supplied in one command message
+    // all actions are executed in parallel.
+    //
+    repeated TrafficAction action = 4;
+
+}
+
+//
+// \brief Atomic Traffic Actions to be performed
+//
+// The actual commands being transmitted are the atomic traffic actions
+// described by this message.
+//
+// \note This message is notionally a multiple choice selection, i.e. only 
+// certain combinations of atomic traffic actions shall be transmitted within 
+// certain time intervals (e.g. for plausibity reasons). The restrictions 
+// regarding that are not part of this message, yet are seen as a task of the 
+// scenario description (e.g. OpenSCENARIO). 
+//
+// \note All TrafficActions are expected to be sent only once, to indicate
+// that they must start immediately. This is also true, if their execution 
+// is expected to take simulation time. To inform the traffic participant 
+// model that certain actions must or shall be terminated, there are 
+// explicit Actions nested inside this message (AbortActionsAction, 
+// EndActionsAction), which hold a reference to the respective actions.
+//
+message TrafficAction
+{
+    // A FollowTrajectoryAction
+    //
+    optional FollowTrajectoryAction follow_trajectory_action = 1;
+
+    // A FollowPathAction
+    //
+    optional FollowPathAction follow_path_action = 2;
+
+    // An AcquireGlobalPositionAction
+    //
+    optional AcquireGlobalPositionAction acquire_global_position_action = 3;
+
+    // A LaneChangeAction
+    //
+    optional LaneChangeAction lane_change_action = 4; 
+
+    // A SpeedAction
+    //
+    optional SpeedAction speed_action = 5;    
+
+    // An AbortActionsAction
+    //
+    optional AbortActionsAction abort_actions_action = 6; 
+
+    // An EndActionsAction
+    //
+    optional EndActionsAction end_actions_action = 7; 
+}
+
+//
```

Comment:
As discussed with @caspar-ai and @Krishna626 this might be a candidate for osi_common.proto (which is imported by this message), because the definition could be re-used in other messages (e.g. TrafficUpdate). TBD prior merging.